### PR TITLE
feat(pre-commit): add base-branch for differential execution

### PR DESCRIPTION
**Pull request by isamu-takagi**
## Description

Add `base-branch` input to pre-commit action for differential check.

## Related links

None

## Tests performed

https://github.com/isamu-takagi/github-actions-sandbox/pull/1

## Notes for reviewers

It needs to add `fetch-depth: 0` to checkout action.

## Pre-review checklist for the PR author

The PR author **must** check the checkboxes below when creating the PR.

- [x] I've confirmed the [contribution guidelines].
- [x] The PR follows the [pull request guidelines].

## In-review checklist for the PR reviewers

The PR reviewers **must** check the checkboxes below before approval.

- [x] The PR follows the [pull request guidelines].
- [x] The PR has been properly tested.
- [x] The PR has been reviewed by the code owners.

## Post-review checklist for the PR author

The PR author **must** check the checkboxes below before merging.

- [ ] There are no open discussions or they are tracked via tickets.
- [ ] The PR is ready for merge.

After all checkboxes are checked, anyone who has write access can merge the PR.

[contribution guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/
[pull request guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/pull-request-guidelines/

### CODE REVIEW

File: pre-commit/README.md [hunk 1]
```diff
@@ -1,7 +1,7 @@
 # pre-commit
 
-This action checks if the PR passes [pre-commit](https://pre-commit.com/).  
-For public repositories, using [pre-commit.ci](https://pre-commit.ci/) is recommended.  
+This action checks if the PR passes [pre-commit](https://pre-commit.com/).
+For public repositories, using [pre-commit.ci](https://pre-commit.ci/) is recommended.
```
Comment (kenji-miyake) [inline pre-commit/README.md:4]: Could you please revert unnecessary changes?

File: pre-commit/action.yaml [hunk 2]
```diff
@@ -18,10 +21,26 @@ runs:
...
         token: ${{ inputs.token }}
 
+    - name: Set target files
+      id: set-target-files
+      run: |
+        if [ -z "${{ inputs.base-branch }}" ]; then
+          echo "option=--all-files" >> $GITHUB_OUTPUT
+          exit 0
+        fi
+        target_files=$(git diff --name-only "${{ inputs.base-branch }}"...HEAD)
```
Comment (kenji-miyake) [inline pre-commit/action.yaml:31]: Is this equivalent to this command? Then it might be simpler.

```bash
pre-commit run --from-ref "${{ inputs.base-branch }}" --to-ref HEAD
```

```
  --from-ref FROM_REF, --source FROM_REF, -s FROM_REF
                        (for usage with `--to-ref`) -- this option represents the original ref in a `from_ref...to_ref` diff expression. For `pre-push` hooks, this represents the branch you are pushing to. For
                        `post-checkout` hooks, this represents the branch that was previously checked out.
  --to-ref TO_REF, --origin TO_REF, -o TO_REF
                        (for usage with `--from-ref`) -- this option represents the destination ref in a `from_ref...to_ref` diff expression. For `pre-push` hooks, this represents the branch being pushed. For
                        `post-checkout` hooks, this represents the branch that is now checked out.
```

File: pre-commit/README.md (hunk 1 above)
Comment (isamu-takagi) [reply pre-commit/README.md]: Done https://github.com/autowarefoundation/autoware-github-actions/pull/237/commits/c5bb04ac9c23c2bd57aefa3e78d8692e1659d969.

File: pre-commit/action.yaml (hunk 2 above)
Comment (isamu-takagi) [reply pre-commit/action.yaml]: Yes. I didn't notice that option. Updated https://github.com/autowarefoundation/autoware-github-actions/pull/237/commits/a6bd4982c27b08ac60a402e0b02ca1d5b137d86c.

File: pre-commit/action.yaml [hunk 3]
```diff
@@ -18,10 +21,25 @@ runs:
       with:
         token: ${{ inputs.token }}
 
+    - name: Set target files
+      id: set-target-files
```
Comment (kenji-miyake) [inline pre-commit/action.yaml:25]: Maybe this?

```suggestion
    - name: Set option
      id: set-option
```

File: pre-commit/action.yaml [hunk 4]
```diff
@@ -18,10 +21,25 @@ runs:
...
+      run: |
+        if [ -z "${{ inputs.base-branch }}" ]; then
+          echo "option=--all-files" >> $GITHUB_OUTPUT
+          exit 0
+        fi
+        echo "option=--from-ref ${{ inputs.base-branch }} --to-ref HEAD" >> $GITHUB_OUTPUT
+      shell: bash
+
+    - name: Get target files
```
Comment (kenji-miyake) [inline pre-commit/action.yaml:34]: I think this step can be removed because it's shown in the next step.

Comment (isamu-takagi) [reply pre-commit/action.yaml]: Removed Updated https://github.com/autowarefoundation/autoware-github-actions/pull/237/commits/00e452a0e130ab2b5331cba8aeecda736151c269.

File: pre-commit/action.yaml [hunk 5]
```diff
@@ -18,10 +21,20 @@ runs:
       with:
         token: ${{ inputs.token }}
 
+    - name: Set option
+      id: set-option
+      run: |
+        if [ -z "${{ inputs.base-branch }}" ]; then
+          echo "option=--all-files" >> $GITHUB_OUTPUT
+          exit 0
```
Comment (kenji-miyake) [inline pre-commit/action.yaml:29]: Is this for debugging?

```suggestion
```

Comment (isamu-takagi) [reply pre-commit/action.yaml:29]: Without this line, the `from-ref` option is specified when the `base-branch` is empty.

Comment (kenji-miyake) [reply pre-commit/action.yaml:29]: Ah, I see. It's a kind of early-return.
Then it's okay. I wasn't used to this style in GitHub Actions.

**Review by kenji-miyake — APPROVED** [review]
Cool, thank you!

**Comment by kenji-miyake** [comment]
@isamu-takagi Released!
https://github.com/autowarefoundation/autoware-github-actions/releases/tag/v1.26.0
